lib/configurationSchema.js

```diff
@@ -0,0 +1,438 @@
+        },
+        {
+          type: 'string',
+        },
+      ],
+    },
+    moduleNameFormatter: {
+      instanceof: 'Function',
+      default: ({ moduleName }) => moduleName,
+    },
+    moduleSideEffectImports: {
+      instanceof: 'Function',
+      default: () => [],
+    },
+    namedExports: {
+      default: {},
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'object',
+          additionalProperties: {
+            type: 'array',
+            items: {
+              type: 'string',
+            },
+          },
+        },
+      ],
+    },
+    packageDependencies: {
+      default: ({ config }) =>
+        findPackageDependencies(
+          config.workingDirectory,
+          config.get('importDevDependencies')
+        ),
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'array',
+          items: {
+            type: `string`,
+          },
+        },
+      ],
+    },
+    parserPlugins: {
+      default: DEFAULT_PARSER_PLUGINS,
+      type: 'array',
+      items: {
+        anyOf: [
+          {
+            type: 'string',
+          },
+          {
+            type: 'array',
+            minItems: 2,
+            additionalItems: false,
+            items: [
+              {
+                type: 'string',
+              },
+              {
+                type: 'object',
+                additionalProperties: {
+                  type: 'string',
+                },
+              },
+            ],
+          },
+        ],
+      },
+    },
+    sortImports: {
+      default: true,
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'boolean',
+        },
+      ],
+    },
+    stripFileExtensions: {
+      default: ['.js', '.jsx', '.ts', '.tsx'],
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'array',
+          items: {
+            type: 'string',
+          },
+        },
+      ],
+    },
+    tab: {
+      default: '  ',
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'string',
+        },
+      ],
+    },
+    useRelativePaths: {
+      default: true,
+      anyOf: [
+        {
+          instanceof: 'Function',
+        },
+        {
+          type: 'boolean',
+        },
+      ],
+    },
+  },
+};
+
+function findGlobalsFromEnvironments(environments) {
+  const result = Object.keys(globals.builtin);
+
+  environments.forEach((environment) => {
+    const envGlobals = globals[environment];
+    if (!envGlobals) {
+      return;
+    }
+    result.push(...Object.keys(envGlobals));
+  });
+  return result;
+}
+
+export function getDefaultConfig() {
+  return Object.entries(SCHEMA.properties).reduce((acc, [k, v]) => {
+    if (typeof v !== `object` || v === null) {
+      throw new Error(`Expected schema key '${k}' to be an object`);
```

Comment:
Instead of saying "key" I think it might make more sense to say "value at" or something similar. It might also be nice to log typeof v.

Reply:
Changed it to 
```
`Expected schema key '${k}' to be an object, but it was of type '${typeof v}'. Got: ${v}`
```

This error will only happen when we change the defaults in code, so it should be fairly obvious what key has changed, but it's always nice to provide as much information as possible.

Reply:
Better but still the wording is a bit off I think. We are talking about the value at that key, not the key itself, but the way it is rded could be understood as referring to the type of the key itself and not its value.